enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,542 @@
+---
+title: multus-cni-for-microshift
+authors:
+  - pmtk
+reviewers:
+  - s1061123, Multus expert
+  - pliurh, Networking expert
+  - dhellmann, MicroShift architect
+  - jerpeter1, Edge Enablement Staff Engineer
+  - pacevedom, MicroShift team lead
+approvers:
+  - dhellmann
+api-approvers:
+  - None
+creation-date: 2024-01-16
+last-updated: 2024-02-06
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-473
+# see-also:
+#   - "/enhancements/this-other-neat-thing.md"
+# replaces:
+#   - "/enhancements/that-less-than-great-idea.md"
+# superseded-by:
+#   - "/enhancements/our-past-effort.md"
+---
+
+# Multus CNI for MicroShift
+
+## Summary
+
+Currently MicroShift ships [ovn-kubernetes](https://github.com/openshift/ovn-kubernetes) (ovn-k)
+CNI responsible for connectivity within and outside the cluster.
+There are users that have needs beyond what ovn-k offers like adding more interfaces to the Pods.
+Some example requirements are connecting Pods to the host's bridge interface or setting up complex networking based on VLAN.
+This functionality is Multus' trademark - adding additional interfaces to Pods.
+
+This enhancement explores providing Multus CNI as an optional component to MicroShift.
+
+## Motivation
+
+Providing Multus CNI for MicroShift will help users:
+- wanting to integrate MicroShift into existing environments
+- wanting to slowly, step by step, migrate to MicroShift without being required to refactor everything at once
+- wanting to add additional interfaces because of newly discovered requirements
+
+### User Stories
+
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
+  make them accessible over networks that should not be available to rest of the cluster.
+* As a MicroShift admin, I want to add additional interfaces to certain Pods so that I can
```

Comment:
I believe that's possible. Actually there's nothing we can do to prohibit that if the NetworkAttachmentDefinition is created like that. I don't know the detail how that would work, but it would be an additional interface - the "original" k8s interface from microshift's ovnk network will continue to be primary interface